### ISSUE TITLE
The base reporter prints the browser object instead of just name

### ISSUE THREAD

**Issue by ramgp** [karma-runner/karma#3305]
https://github.com/karma-runner/karma/blob/c311ac00cef4a957f889ed613238793c0ffa95a2/lib/reporters/base.js#L40

Expected `HeadlessChrome 75.0.3765 (Linux 0.0.0)` and got:
```
Browser {
  id: '72739917',
  fullName: 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 ' +
    '(KHTML, like Gecko) HeadlessChrome/75.0.3765.0 ' +
    'Safari/537.36',
  name: 'HeadlessChrome 0.0.0 (Linux 0.0.0)',
  state: 1,
  lastResult: [Result],
  disconnectsCount: 0,
  init: [Function],
  isReady: [Function],
  toString: [Function],
  onKarmaError: [Function],
  onInfo: [Function],
  onStart: [Function],
  onComplete: [Function],
  onDisconnect: [Function],
  reconnect: [Function],
  onResult: [Function],
  serialize: [Function],
  execute: [Function]
}

**Comment by devoto13** [comment]
`Browser` class has a `toString()` method, so it should print the name already. And it does so for me, when using Node 8.16.0 and `master` branch. Can you provide more information about versions of Node/Karma you're using and your environment, so we can reproduce the issue?

**Comment by ramgp** [comment]
I don't know the specific versions where it started appearing, I'm gonna guess it started with Node 11.x and Browser 74+ The current Node version is up to 12.x now and Browser is `HeadlessChrome 75.0.3765 (Linux 0.0.0)`. I'm using a different reporter `spec` as a workaround to the problem, because it only prints the Browser object once at the end.

**Comment by devoto13** [comment]
Thanks! This was helpful. 

Karma uses `util.format()` function from Node to print instance of the `Browser` and this function changed behaviour somewhere between Node 10 and 12.

- This will use `browser.toString()` and output only name: https://nodejs.org/docs/latest-v10.x/api/util.html#util_util_format_format_args
- This will use `util.inspect()` and output whole browser object: https://nodejs.org/dist/latest-v12.x/docs/api/util.html#util_util_format_format_args

**Comment by falsandtru** [comment]
Also see #3322

**Comment by robertknight** [comment]
This appears to be a bug/issue in Node itself. The `Browser` class defines a `toString` method and the current master branch of Node will use that. See https://github.com/nodejs/node/commit/5518664d41b8916dfbe2eca2180d760db632748e#diff-91f6ca3fae2551d5959872b74c18946a

**Comment by robertknight** [comment]
The following code run in the Node REPL prints the full `Browser` object in Node v12.1.0 but the browser name in Node v12.4.0:

```js
var Browser = require('karma/lib/browser')
var b = new Browser
b.name = 'test'
util.format('%s', b) // Prints full object in Node v12.1.0 and "test" in Node v12.4.0.
```

**Comment by falsandtru** [comment]
Should be closed?

**Comment by davidje13** [comment]
I can confirm that having observed this issue with node 12.1.0, it is resolved after upgrading to node 12.4.0
